dsn/dsn.go

```diff
@@ -113,21 +113,26 @@ func (dsn DSN) AutoDetect() (DSN, error) {
 }
 
 func Defaults(defaultsFile string) (DSN, error) {
-	var params []string
+	versionParams := [][]string{}
+	versionParams = append(versionParams, []string{}) // [0] = params for MySQL 5.6+
+	versionParams = append(versionParams, []string{}) // [1] = params for MySQL 5.5
 	if defaultsFile != "" {
-		params = []string{
-			"--defaults-file=" + defaultsFile,
-			"--print-defaults", // --print-defaults must be last param
-		}
-	} else {
-		params = []string{
-			"--print-defaults",
-		}
+		versionParams[0] = append(versionParams[0], "--defaults-file="+defaultsFile)
+		versionParams[1] = append(versionParams[1], "--defaults-file="+defaultsFile)
 	}
-	cmd := exec.Command("mysql", params...)
-	output, err := cmd.Output()
-	if err != nil {
-		return DSN{}, err
+
+	versionParams[0] = append(versionParams[0], "-s")
+	versionParams[0] = append(versionParams[0], "client")
+	versionParams[1] = append(versionParams[1], "client")
+
```

Comment:
It works however I suggest to simplify this code for better understanding like this:
``` 
versionParams := [][]string{
[]string{"--defaults-file="+defaultsFile, "-s", "client"},
[]string{"--defaults-file="+defaultsFile, "client"},
}
```